crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -855,4 +867,64 @@ where
             pipe_from_stream(pending, mapping_stream).boxed(),
         );
     }
+
+    fn subscribe_filtered_object_mappings(
+        &self,
+        pending: PendingSubscriptionSink,
+        hashes: Vec<Blake3HashHex>,
+    ) {
+        if hashes.len() > MAX_OBJECT_HASHES_PER_SUBSCRIPTION {
+            warn!(
+                "Request hash count ({}) exceed the server limit: {} ",
+                hashes.len(),
+                MAX_OBJECT_HASHES_PER_SUBSCRIPTION
+            );
+
+            let err_fut = pending.reject(ErrorObject::owned(
+                // TODO: is there a defined error code for an internal request limit?
+                1,
+                format!(
+                    "Request hash count ({}) exceed the server limit: {} ",
+                    hashes.len(),
+                    MAX_OBJECT_HASHES_PER_SUBSCRIPTION
+                ),
+                None::<()>,
+            ));
+
+            self.subscription_executor.spawn(
+                "subspace-filtered-object-mappings-subscription",
+                Some("rpc"),
+                err_fut.boxed(),
+            );
+
+            return;
+        };
+
+        let hashes: HashSet<Blake3Hash> = HashSet::from_iter(hashes.into_iter().map(|hash| *hash));
```

Comment:
Yes. The Deref and Into impls are ambiguous, because Blake3Hash is a type alias to an array.

The best we can do is something like:
```suggestion
        let hashes = HashSet::from_iter(hashes.into_iter().map(|hash| Blake3Hash::from(hash)));
```

Reply:
Or this, which is typically the shortest:
```suggestion
        let hashes = HashSet::<Blake3Hash>::from_iter(hashes.into_iter().map(|hash| *hash));
```